tests/unit/deferred.py

```diff
@@ -0,0 +1,201 @@
+# License. You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS
+# IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+'''Functional test for deferred.py'''
+
+import pytest
+import os
+import tempfile
+
+from hil import config, deferred, model
+from hil.model import db, Switch
+from hil.test_common import config_testsuite, config_merge, \
+                             fresh_database, fail_on_log_warnings
+from flask import Flask
+from flask_sqlalchemy import SQLAlchemy
+
+fail_on_log_warnings = pytest.fixture(autouse=True)(fail_on_log_warnings)
+fresh_database = pytest.fixture(fresh_database)
+
+INTERFACE1 = '104/0/10'
+INTERFACE2 = '104/0/18'
+
+DeferredTestSwitch = None
+temp_db = tempfile.NamedTemporaryFile()
+
+
+@pytest.fixture
+def configure():
+    config_testsuite()
+
+    additional_config = {
+        'extensions': {
+            'hil.ext.obm.mock': ''
+            }
+        }
+
+    # if we are using sqlite's in memory db, then change uri to a db on disk
+    uri = config.cfg.get('database', 'uri')
+    if uri == 'sqlite:///:memory:':
+        uri = 'sqlite:///' + temp_db.name
+        additional_config['database'] = {'uri': uri}
+
+    config_merge(additional_config)
+    config.load_extensions()
+
+
+@pytest.fixture()
+def _deferred_test_switch_class():
+    global DeferredTestSwitch
+
+    class DeferredTestSwitch_(Switch):
+        '''DeferredTestSwitch
+
+        This is a switch implemented to test the deferred.apply_networking()
+        function.  It is needed for a custom implementation of the switch's
+        apply_networking() that counts the networking actions as
+        apply_networking() is called to ensure it is incremental.
+        '''
+
+        api_name = 'http://schema.massopencloud.org/haas/v0/switches/deferred'
+
+        __mapper_args__ = {
+            'polymorphic_identity': api_name,
+        }
+
+        id = db.Column(db.Integer,
+                       db.ForeignKey('switch.id'),
+                       primary_key=True)
+        hostname = db.Column(db.String, nullable=False)
+        username = db.Column(db.String, nullable=False)
+        password = db.Column(db.String, nullable=False)
+        last_count = None
+
+        @staticmethod
+        def validate(kwargs):
+            pass
+
+        def session(self):
+            return self
+
+        def disconnect(self):
+            pass
+
+        def modify_port(self, port, channel, network_id):
+
+            # setup a different connection to database so that this method does
+            # not see uncommited changes by `apply_networking`
+
+            local_app = Flask(__name__.split('.')[0])
```

Comment:
Do you know of an easier way of getting a separate flask context than doing all of this inline?

Maybe [app_context()](http://flask.pocoo.org/docs/0.10/appcontext/)?